airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_insight_streams.py

```diff
@@ -152,9 +160,11 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
     def _date_intervals(self) -> Iterator[pendulum.Date]:
         """Get date period to sync"""
-        if self._end_date < self._next_cursor_value:
+        today = pendulum.today(tz="UTC").date()
+        end_date = min(self._end_date, today)
```

Comment:
Perhaps since we are going to prevent looking at "today" again once that date has already been looked up, the earliest day we look at should be "yesterday" (e.g. today - 1).  This way, we won't try to look up data for a partial day without being able to update it.

Reply:
I have changed logic to sync only for yesterday